### DISCUSSION TITLE
export .3mx to .obj

### DISCUSSION THREAD

**Discussion by julesmorel** [openscenegraph/OpenSceneGraph#1184] [Q&A] [answered]
I've been using the following plugin to load Bentley files .3mx into openSceneGraph:
https://github.com/ProjSEED/osgPlugins-3mx

The .3mx file format has a LOD structure, consisting of several surface models displayed according to the camera position and zoom.

Thanks to the plugin, I display my files correctly using osgviewer. However, I am struggling to convert those models to .obj format using osgconv. Are there any options to set up as it is a LOD structure?

**Comment by robertosfield** [comment]
.obj format doesn't have LOD support so it;'s impossible to write a scene graph containing LOD's to .obj.

Rather than try and chase the impossible select an exchange format that supports LODs.

**Reply by julesmorel** [reply]
I was thinking it would be possible to export .obj at the smallest resolution contained in the 3mx.

Could you point another format that supports LOD?

**Reply by robertosfield** [reply]
Could you say what purpose you are trying to convert these files?  Without that information we can't really say what might suit your purpose.

As a general comment, the OpenScenGraph is a scene graph library, it's not a data exchange utility.  osgconv primary function is for converting files to native OSG formats.

**Reply by julesmorel** [reply]
I have access to a collection of photogrammetry models stored in the format .3mx

I'd like to import those models into blender to render them. As the 3mx format is not supported by blender, I need to first convert them and OSG appeared to be the only option capable of doing it by using the custom plugin I was previously talking about.

Of course, if you have in mind any other idea to import my models into blender, that would be perfect

**Comment by andek714** [comment]
OpenFlight (.flt), Collada (.dae) or Autodesk FBX (.fbx) support LODs and can be written by their OSG plugins. But if I understand correctly, this format is used for storing very high-res models in a quad-tree och octree LOD structure? If so, a file with all levels might be impossibly large to handle for another tool, without the PagedLOD support in OSG.

If you just want to save the highest levels you can maybe traverse down to those leaf nodes and save to separate .obj if you want.

I'm not familiar with the format, but maybe the highest levels are already separate .3mx files on disk that you can convert?

**Reply by julesmorel** [reply]
Indeed, highest levels are separate 3mx files. However, the export of a single of those files results in an .obj which correctly describes the shape, but it's texture is missing for some reason. The corresponding .mtl file is produced, but it is empty.

**Comment by julesmorel** [comment, marked as answer]
I solved my problem by converting first to .osg for the desired LOD, then .osg to .obj :
`osgconv -O OutputTextureFiles -e 3mx intput_file.3mx output_file.osg`
` osgconv output_file.osg output_file.obj `

The material files produced contained links to .dds textures. I converted those textures to .jpg files, then changed the path into each .mtl files.
